tests/appsec/test_shell_execution.py

```diff
@@ -38,29 +43,27 @@ def setup_track_cmd_exec(self):
     )
     def test_track_cmd_exec(self):
         shell_exec_span = self.fetch_command_execution_span(self.r_cmd_exec)
+        assert shell_exec_span["resource"] == "echo"
         assert shell_exec_span["meta"]["cmd.exec"] == '["echo","foo"]'
         assert shell_exec_span["meta"]["cmd.exit_code"] == "0"
 
     def setup_track_shell_exec(self):
         self.r_shell_exec = weblog.post(
-            "/shell_execution",
-            headers={"Content-Type": "application/json"},
-            data=json.dumps({"command": "echo", "options": {"shell": True}, "args": "foo"}),
+            "/shell_execution", json={"command": "echo", "options": {"shell": True}, "args": "foo"},
         )
 
     @irrelevant(library="java", reason="No method for shell execution in Java")
     def test_track_shell_exec(self):
         shell_exec_span = self.fetch_command_execution_span(self.r_shell_exec)
-
+        # Some implementations will compute the shell command as "sh -c echo foo".
+        assert shell_exec_span["resource"] in ("echo", "sh")
```

Comment:
From my reading of the spec, the only possibilities here:

* resource=sh, cmd.shell=echo foo
* resource=sh, cmd.exec=["sh", "-c", "echo foo"]
* resource=echo, cmd.exec=["echo", "foo"], if the echo binary is executed directly

Since this test is about shell execution, we can exclude the third. Arguably, the second is not appropriate for this test either, but in any case what doesn't make sense is to have resource=echo, cmd.shell=echo foo. The spec says:

> If a shell command is being executed (i.e. cmd.shell is populated), the resource is the name of the shell used (default to sh).

Reply:
That's correct. cc @hoolioh Node's implementation is not consistent with this. Is it ok if I mark this test as `@bug` for it?

Reply:
I'm looking into it. Anyway Shell Execution feature is not merged yet in the nodejs tracer so I think you don't need to annotate the test.